cmd/run.go

```diff
@@ -61,7 +64,16 @@ export const main = async (...args: string[]) => {
 
 		runOpts.Args = args[1:]
 
-		err = internal.Run(repo, runOpts)
+		ctx, cancel := context.WithCancel(cmd.Context())
+		shutdown := make(chan os.Signal, 1)
+		signal.Notify(shutdown, syscall.SIGINT)
+		go func() {
+			_ = <-shutdown
+			fmt.Fprintf(os.Stderr, "Received shutdown signal. Waiting for process to finish.\n")
+			cancel()
```

Comment:
There are two things going on here that are not "library-safe", but I think it's OK because this is "cmd" code and not "internal" code.

1) I believe you're supposed to `defer cancel()` right after you call `WithCancel`, so that you can cleanup any channel resources that are allocated in order to make cancelling possible.
2) This go routine may block on `shutdown` forever.

Not sure there is any action for this comment, just thinking aloud as I review this code.